### PR TITLE
Implement loom native methods

### PR DESCRIPTION
**Pull request by JasonFengJ9**
```
JVM_IsPreviewEnabled
Java_java_lang_Thread_setCurrentThread
Java_java_lang_Thread_extentLocalCache
Java_java_lang_Thread_setExtentLocalCache
```

Related to https://github.com/eclipse-openj9/openj9/issues/15176

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: runtime/j9vm/javanextvmi.c [hunk 1]
```diff
@@ -219,8 +219,12 @@ JVM_IsContinuationsSupported(void)
 JNIEXPORT jboolean JNICALL
```
Comment (tajila) [inline runtime/j9vm/javanextvmi.c:219]: you can remove the assertion for `Jenkins test sanity xLinux jdk11` and just return false

File: runtime/jcl/common/thread.cpp [hunk 2]
```diff
@@ -417,22 +417,32 @@ Java_java_lang_Thread_currentCarrierThread(JNIEnv *env, jclass clazz)
 void JNICALL
 Java_java_lang_Thread_setCurrentThread(JNIEnv *env, jclass clazz, jobject thread)
 {
-	Assert_JCL_unimplemented();
+	J9VMThread *currentThread = (J9VMThread*)env;
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	currentThread->threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
```
Comment (tajila) [inline runtime/jcl/common/thread.cpp:422]: you need to use barrier APIs for this

File: runtime/jcl/common/thread.cpp [hunk 3]
```diff
@@ -417,22 +417,32 @@ Java_java_lang_Thread_currentCarrierThread(JNIEnv *env, jclass clazz)
...
 /* static native Object[] extentLocalCache(); */
-jobject JNICALL
+jobjectArray JNICALL
 Java_java_lang_Thread_extentLocalCache(JNIEnv *env, jclass clazz)
 {
-	Assert_JCL_unimplemented();
-	return NULL;
+	J9VMThread *currentThread = (J9VMThread*)env;
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	jobjectArray result = (jobjectArray)VM_VMHelpers::createLocalRef(env, currentThread->extentLocalCache);
```
Comment (tajila) [inline runtime/jcl/common/thread.cpp:432]: same here and below

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
@dmitripivkine please review

File: runtime/j9vm/javanextvmi.c (hunk 1 above)
Comment (JasonFengJ9) [reply runtime/j9vm/javanextvmi.c:222]: done

File: runtime/jcl/common/thread.cpp (hunk 2 above)
Comment (JasonFengJ9) [reply runtime/jcl/common/thread.cpp:419]: Assuming `issueWriteBarrier()` is needed here, wondering what's the use scenario/concern requiring this perf sensitive call?

File: runtime/jcl/common/thread.cpp (hunk 3 above)
Comment (JasonFengJ9) [reply runtime/jcl/common/thread.cpp:429]: Similar question, assuming `issueReadBarrier()` is needed, any particular scenario require it?

**Review by dmitripivkine — APPROVED** [review]

File: runtime/jcl/common/thread.cpp (hunk 2 above)
Comment (tajila) [reply runtime/jcl/common/thread.cpp:420]: I meant GC barrier, ie J9STATIC_OBJECT_STORE, J9STATIC_OBJECT_LOAD

Comment (amicic) [reply runtime/jcl/common/thread.cpp:420]: Storing into or reading from references in a native structure that acts as a _strong_ root (such as J9VMThread, thread slots etc), does not require object assess barrier. 
It would be nice to use it, for potential tracking/debugging, and since in theory there are GCs that care about root slot accesses, but current we don't have any GC that does. 
We don't even have an API for it, although J9STATIC_OBJECT_STORE/LOAD could be 'abused' with providing class value NULL, but if we ever start doing it, we should really have dedicated APIs like J9STRONGROOT_OBJECT_STORE/LOAD

We require object access barriers for:
- heap objects slots, including their native struct extension that do not reside in heap. Hence we have J9STATIC_OBJECT_STORE/LOAD for class object slots extensions, aka class structs. 
- reading reference slots for native struct that act as _weak_ root (specifically, Monitor and String tables), and API is J9WEAKROOT_OBJECT_LOAD

In short, this line looks ok to me.

Comment (amicic) [reply runtime/jcl/common/thread.cpp:420]: Access to reference fields in VirtualThread native struct (Continuation or whatever), since they are not roots would require barrier, though.
